HDS-1374 docs navigation mobile style fixes

mikkojamG opened the pull request:
## Description

<!-- Describe your changes in detail -->
<!-- Add [Feature] or [BreakingChange] to the title -->

Fixing docs navigation overlapping menu button on smaller screen sizes (320px). Also made some small changes so that titles longer than "Helsinki Design System" should work a little bit better on smaller screens.

## Related Issue

<!-- If suggesting a new feature or change, please discuss it in an issue first -->
<!-- If fixing a bug, please link to the issue here: -->

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1374

## Motivation and Context

<!-- Why is this change required? What problem does it solve? -->

Overlapping elements will eventually cause poor user experience.

## How Has This Been Tested?
- Unit testing
- Local testing
- Testing on different browsers (Edge emulated)

## Screenshots (if appropriate):

<img width="484" alt="Screenshot 2023-02-06 at 13 35 58" src="https://user-images.githubusercontent.com/66477579/216961632-0e6da84a-dcbe-459f-be3a-a7de6470bec3.png">

<img width="432" alt="Screenshot 2023-02-06 at 13 36 13" src="https://user-images.githubusercontent.com/66477579/216961654-6279163f-b799-4a38-a4cd-880ad0b3b9cf.png">

minevala commented:
It seems that the Navigation component has also changed. At least some Navigation loki-test reference images do not match the implementation 🤔 
https://github.com/City-of-Helsinki/helsinki-design-system/actions/runs/4103234489/jobs/7078235174#step:5:18

mikkojamG commented:
> It seems that the Navigation component has also changed. At least some Navigation loki-test reference images do not match the implementation 🤔 https://github.com/City-of-Helsinki/helsinki-design-system/actions/runs/4103234489/jobs/7078235174#step:5:18

Fixed 🙂

laurakarhu reviewed (approved):
Looks good to me :)
